### PR TITLE
Utku / Class 19 / Homework Week 2

### PR DESCRIPTION
**Pull request by u61**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,15 +15,162 @@
     return elem;
   }
 
+  function remove() {
+    document.getElementById('right_column_list').remove();
+  }
+
+  function setLink(linkID, srcArray) {
+    document.getElementById(linkID).innerText = srcArray.name;
```
Comment (yash-kapila) [inline homework/index.js]: If you look closely, you are trying to fetch the element here in all three steps before you assign its `innerText`, `href` and `target` attributes. You can instead have the calling function i.e from where we call `setLink` to provide the element itself or simply retrieve it once in this function and use it in all three assignments. This way, you wouldn't have to fetch the same element thrice and will save yourself two operations. 

```
function setLink(linkId, srcArray) {
  const elem = document.getElementById(linkID);
  elem.innerText = srcArray.name;
  elem.setAttribute('href', srcArray.html_url);
  elem.setAttribute('target', '_blank');
}
```

File: homework/index.js [hunk 2]
```diff
@@ -30,15 +15,162 @@
...
+        select.addEventListener('change', () => {
+          const selected = select.selectedIndex;
+          const fetchedDataContributors = fetch(contributorsURL[selected]).then(response =>
+            response.json(),
+          );
+          const rigthColumnList = createAndAppend('ul', rightPanel, {
+            id: 'right_column_list',
+          });
+
+          fetchedDataContributors
```
Comment (yash-kapila) [inline homework/index.js]: Lines 131 to 166 can hence be removed when we create a common function for fetching contributors. 🙂

File: homework/index.js [hunk 3]
```diff
@@ -30,15 +15,162 @@
...
+                text: contributor.contributions,
+              });
+            });
+          });
+        }
+
+        createRightPanel();
+
+        select.addEventListener('change', () => {
+          remove();
```
Comment (yash-kapila) [inline homework/index.js]: You can also put the `remove()` inside the listener you are having below.

File: homework/index.js [hunk 4]
```diff
@@ -30,15 +15,162 @@
...
+    document.getElementById('right_column_list').remove();
+  }
+
+  function setLink(linkID, srcArray) {
+    document.getElementById(linkID).innerText = srcArray.name;
+    document.getElementById(linkID).setAttribute('href', srcArray.html_url);
+    document.getElementById(linkID).setAttribute('target', '_blank');
+  }
+
+  const root = document.getElementById('root');
```
Comment (yash-kapila) [inline homework/index.js]: I would suggest you move all these statements to its own function which you can call from the `main()`. And if you are using some of these elements in other places, you can then try return those elements or do a `document.getElementById` in areas where you need them.

File: homework/index.js [hunk 5]
```diff
@@ -30,15 +15,162 @@
...
+          document.getElementById('property_updated').innerText = new Date(dataIndex.updated_at);
+        }
+
+        assignLeftPanelValues(data[0]);
+
+        function createRightPanel() {
+          const rigthColumnListDefault = createAndAppend('ul', rightPanel, {
+            id: 'right_column_list',
+          });
+          fetchedDataContributorsDefault.then(data2 => {
```
[the remark this answers is not in the document]
Comment (u61) [reply homework/index.js]: Created a new function and implemented it required places as you mentioned.

File: homework/index.js (hunk 1 above)
Comment (u61) [reply homework/index.js]: Replaced your code but I understand what you meant.
Surely, this should be like this.

File: homework/index.js (hunk 2 above)
Comment (u61) [reply homework/index.js]: Yes, I removed that huge block.

File: homework/index.js (hunk 3 above)
Comment (u61) [reply homework/index.js]: Absolutely. I failed to noticed that as well.

File: homework/index.js (hunk 4 above)
Comment (u61) [reply homework/index.js]: Actually, I was trying to empty main function in order to get more pure code. But, in the end there is no need for this. Included them again.

**Comment by u61** [comment]
Hello Yash,

I have tried to refactor my code as per your suggestions.

Thank you for your corrections and your time.

**Pull request closed without merging**